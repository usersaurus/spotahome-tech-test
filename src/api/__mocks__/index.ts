import { IRawHouseData } from '../../bff'

export const getHouses = () =>
  Promise.resolve([
    {
      Address: 'Paseo de la Castellana',
      City: 'madrid',
      CountryCode: 'es',
      Price: 280000,
      StartDate: '2018-11-04',
      EndDate: '2018-12-20',
      Currency: 'EUR',
      Description:
        'This luxury 3 bedroom apartment can accommodate up to six people. The French pool table is the centerpiece of this spacious 140 m2 penthouse apartment with views over Paseo de la Castellana. The comfortable apartment has air conditioning in the living room and main bedroom, central heating, a fully equipped kitchen, a 42\u201d plasma TV, DVD player, stereo, Nintendo Wii + Wii Sports, and an Apple TV.\\n- The home was professionally decorated using furniture from Charles Eames and Le Corbusier.\n- The living room is very large and has three different areas. One area with sofas and entertainment system, a dining area with a designer glass table and chairs, and an area with a full size French pool table.\n- The large balcony has views of Paseo de la Castellana.\\n\\nThe home is very well connected with the metro station Plaza Castilla and several bus lines.  It is also just a few meters away from an entrance to the M30, the highway that circles the city, as well as the Chamartin train station. From there you can catch the "Cercanias" commuter trains that go all over the Province of Madrid.\\n\\n\\n',
      Lat: 40.469542599999997,
      Lng: -3.6879847999999811,
      Facilities: {
        ac: 'yes',
        tv: 'yes',
        bed: 'yes',
        desk: 'no',
        lock: 'yes',
        shed: 'no',
        wifi: 'yes',
        dryer: 'no',
        closet: 'no',
        garden: 'no',
        toilet: 'private',
        balcony: 'private',
        bedroom: 'private',
        heating: 'central',
        kitchen: 'private',
        parking: 'no',
        terrace: 'private',
        basement: 'no',
        bathroom: 'private',
        flooring: 'laminate',
        internet: 'yes',
        dishwasher: 'yes',
        play_music: 'discussable',
        kitchenware: 'private',
        living_room: 'private',
        animal_allowed: 'no',
        cleaning_common: 'no',
        lroom_furniture: 'no',
        smoking_allowed: 'no',
        washing_machine: 'yes',
        allergy_friendly: 'no',
        cleaning_private: 'no',
        wheelchair_accessible: 'no',
      },
      Deposit: 280000,
      Images: [
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/74ef00db848c1ce44ea03677836b5bb61243e3f9d6a9d5a656d3c3cb.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/f4099965ba78e0c376a1752c7187d779878da3182eb170c6a116273d.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/62dcda859e73f911ad965ab7374957d3c2a57314efbcda7566bf4410.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/c83fdc4a890444e1f201b500fca5ed1b0cf8159205104db57e7764a7.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/ee3f1932c7efdf4990662e8ef9e1e2bf8149242e3631e211150ec40d.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/46722effec98b3d431cf5f21ff4c3e9689a3606040a5c489233a359d.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/3afac2d145d329b4d0857f2d9ad260cf43cf8fe227b30e7421669650.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/33324c6a437f1455840175ad4d5a7e2f3d7d8e093697ff188b2fc858.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/b5f9f5911a07bdfe7ec84366e42670c51f63e20192ee17952cca3fe9.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/69898435b30a34f8aa7fab934864025c96596877ec857246927e1d7c.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/99075214f75f3fff38d35ddf84929fdbfb09ff368850649db9b1fddf.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/47d34632281d7ea943d8f787eeda45ace2f30d14d0b88623a247ebed.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/d7c70fb12920ccecac7421013c5c4ba1e228ad1f0a3629a7cccdf4b5.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/fcbb11f000732f646f29c10cb30d55ed396f8e53ad6db206e47c468c.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/fce15ea0d4d7c2d6d696f733469df93523258042f85c7e7b3e4802f0.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/16962d14605399e16e6ffd371a5eaaf5f6fcbc17b2c20149b71f8d3f.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/ebe8033557ba26a0228b9f96fd5c4424964af54dc2c3813db84f3f30.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/f44377665eceb0882f4e0105a85d59ebb118473e0b20228c0df63b8a.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/545306877ada1b3b9a7120a2774c291f9f2fb219a649fcb11bed25e3.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/2ef5176a466be4bbb9def87f3126ad9f120e5118fd1cac1dca344287.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/43bbc8b5756c31f77a6e79df0005c6aa96ab7239550c27b1358d4d59.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/7d05c4f1d992f7428549536dbbeb1fea381f39631b5f2b25b348027e.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/6b929f848b89322d9cc8125989be836c2e7ba47a5f3498a395e306fa.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/a6d207a3ad103fd5e9c90ef651ded98ea986806b73aa34932a181724.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/f2326835a0037f433dde5c272e816ab41f8783a55f1eec768245fe8d.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/ba3602602aefddd408186394a66fc1f6fa1437fd8e1d24eb25bae346.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/864bce784e2bf97c05863bb2f1120fd9e0da91d83f6f5d2940e4f6dc.jpg',
      ],
      LastUpdated: '2018-10-10',
      ListingReference: '79906',
      Link: 'https://www.spotahome.com/madrid/for-rent:apartments/79906?utm_source=housinganywhere&utm_medium=cffeeds&utm_campaign=normalads',
    },
    {
      Address: 'calle virtudes',
      City: 'madrid',
      CountryCode: 'es',
      Price: 95000,
      StartDate: '2019-11-01',
      EndDate: '',
      Currency: 'EUR',
      Description:
        'This 45 square meter, one bedroom apartment is located in one of Madrid\u2019s most authentic neighbourhoods \u2013 Chamber\u00ed. The apartment is within walking distance to transportation, as well as many pubs, restaurants, and shops. \n\nWhen you enter the apartment there is a corridor. The first room on your right is the kitchen. The next room off of the corridor is an interior patio. The corridor opens up into the living room, which then continues on to another section where the bedroom and bathroom are located.\n\nThe kitchen is equipped with a microwave, electric stove, refrigerator, amenities, and storage space. In the interior patio, there is a washing machine as well as the water heater, which is run on natural gas. The living room has a dining table with chairs and a sofa. The bedroom comes with a double bed, a night table, and a built-in wardrobe. The bedroom has a window which looks onto the interior patio. The bathroom has a sink, a shower and bath, and a toilet.\\nThe area is pedestrian-friendly.\\nRequires TWO MONTH deposit.\\nLocated just a block away from the apartment, is the metro station of Alonso Cano, which serves line 7. There is also  the metro station of Iglesia, which is a short walk away, serving line 1. There are also local bus routes serving lines 3 and 12.\\n\\n\\n',
      Lat: 40.437862593700004,
      Lng: -3.6987319177999911,
      Facilities: {
        ac: 'yes',
        tv: 'no',
        bed: 'yes',
        desk: 'no',
        lock: 'yes',
        shed: 'no',
        wifi: 'no',
        dryer: 'no',
        closet: 'no',
        garden: 'no',
        toilet: 'private',
        balcony: 'no',
        bedroom: 'private',
        heating: 'no',
        kitchen: 'private',
        parking: 'yes',
        terrace: 'no',
        basement: 'no',
        bathroom: 'private',
        flooring: 'laminate',
        internet: 'no',
        dishwasher: 'no',
        play_music: 'discussable',
        kitchenware: 'no',
        living_room: 'private',
        animal_allowed: 'no',
        cleaning_common: 'no',
        lroom_furniture: 'yes',
        smoking_allowed: 'no',
        washing_machine: 'yes',
        allergy_friendly: 'no',
        cleaning_private: 'no',
        wheelchair_accessible: 'no',
      },
      Deposit: 95000,
      Images: [
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/9617b48286d8cf89fe755b19d884901be10f6b613a6fc872be2e83be.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/c33a5bdc5a827b95b3ab9b6c6b8f02ebb9210a4aae77f4d27dc10ef7.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/ec637fc352df855f8fd3f18c2c85e250708e1318b7d48a79c6702a01.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/018310f31e20e952730ebe25494b720e17dee4d876fd9858eac94fcc.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/3c29f8051b37b51686e535dea5e03a4bffb45629ac3d5f8b695a23a3.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/065e4fb815430e526582cdf8840c301654a736fb4372047bac312bb8.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/b523375ec89908aeff9ee990f891c28da53bf616b869ab8fcfcb2ee6.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/d0b62ae38a968872e3ab121113839f1c2add9214aebf9b62fb58e4c5.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/e1ecc9ffee88b98d04d2f75070e9022d12bd6c54a6fb951cda612d27.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/20db69b32c525acdacbbfe878d1f30912ca4dad07acab8462b957f7f.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/a493b014eecb2876d56f46c773c28496503338dd7a41e467aa47f722.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/841af9b8e5ba90d740a3fff749d62f8f20eb2df62ed6cf25be9be03e.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/c463ee46dff122415c4ad88c29048ebb7c2e23b2876042e4292fb83d.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/6b30b56be1d8555763c8ab930a1915f3c7e5ef86552f79f1446bc8e6.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/b5b70ea7e48c7ccd3abf431b39bb330442476e6b5612c0f46a93748d.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/65b056040d5672be33a896fd86ff59da544c963427c07e0020da2888.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/25b60d832d198252d159a23a6804caf8d92e2fa6e8e2f3200b6c756c.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/31703ceb7d6a10b863eba256ebdb022e89f7aafb28991dd763125736.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/1ec15f44a76e6668bd869b8327036389455646ab3b67fcb3a363e00d.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/ac4280cb34b20e0445fe40cc27f2658b5450b5e6bf125da06bbd07ce.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/a923f2b91ca1ab1cdf84b5339fb191c4e9304516d175664a2bdabb0b.jpg',
        'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/4afa64c7675675023f020a565a23f22c9a18b5622e4d41d7c34cc708.jpg',
      ],
      LastUpdated: '2018-08-03',
      ListingReference: '79994',
      Link: 'https://www.spotahome.com/madrid/for-rent:apartments/79994?utm_source=housinganywhere&utm_medium=cffeeds&utm_campaign=normalads',
    },
  ] as IRawHouseData[])
