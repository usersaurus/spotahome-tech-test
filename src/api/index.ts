import axios from 'axios'
import { IRawHouseData, IPagination } from '../bff'

export const getHouses = (page: number) =>
  axios
    .get<{ pagination: IPagination; data: IRawHouseData[] }>(
      'http://localhost:3000/gethouses',
      { params: { page } }
    )
    .then(({ data }) => data)
