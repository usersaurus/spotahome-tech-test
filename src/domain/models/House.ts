import { IRawHouseData } from '../../bff'

export interface IHouse {
  id: string
  title: string
  address: string
  city: string
  link: string
  image: string
}

export const House = (props: IRawHouseData) =>
  ({
    id: props.ListingReference,
    title: `${props.City} - ${formatCurrency(
      props.CountryCode,
      props.Currency
    ).format(props.Price)}`, // There's not a title prop in given data, so we'll just use the city and price
    address: props.Address,
    city: props.City,
    link: props.Link,
    image: props.Images.length && props.Images[0],
  } as IHouse)

const formatCurrency = (countryCode: string, currency: string) =>
  new Intl.NumberFormat(countryCode, {
    style: 'currency',
    currency,
  })
