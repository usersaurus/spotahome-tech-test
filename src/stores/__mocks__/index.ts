import { IHouse } from '../../domain/models/House'

export const houseList = [
  {
    title: 'madrid - 280000',
    link: 'https://www.spotahome.com/madrid/for-rent:apartments/79906?utm_source=housinganywhere&utm_medium=cffeeds&utm_campaign=normalads',
    address: 'Paseo de la Castellana',
    city: 'madrid',
    image:
      'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/74ef00db848c1ce44ea03677836b5bb61243e3f9d6a9d5a656d3c3cb.jpg',
  },
  {
    title: 'madrid - 95000',
    link: 'https://www.spotahome.com/madrid/for-rent:apartments/79994?utm_source=housinganywhere&utm_medium=cffeeds&utm_campaign=normalads',
    address: 'calle virtudes',
    city: 'madrid',
    image:
      'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/9617b48286d8cf89fe755b19d884901be10f6b613a6fc872be2e83be.jpg',
  },
  {
    title: 'madrid - 165000',
    link: 'https://www.spotahome.com/madrid/for-rent:apartments/79988?utm_source=housinganywhere&utm_medium=cffeeds&utm_campaign=normalads',
    address: 'Calle Ruda',
    city: 'madrid',
    image:
      'https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/64843eb86338aef2ed4420a3eb7f49cc7a9267aee3392de6979b0193.jpg',
  },
] as IHouse[]
