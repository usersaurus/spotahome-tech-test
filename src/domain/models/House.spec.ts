import { House } from './House'
import { getHouses } from '../../api/__mocks__'

describe('The House model', () => {
  it('matches snapshot when object is created with IRawHouseData info', async () => {
    const housesListInfo = await getHouses()
    const house = House(housesListInfo.data[0])

    expect(house).toMatchSnapshot()
  })
})
