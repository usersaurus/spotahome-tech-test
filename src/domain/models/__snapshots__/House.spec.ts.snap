// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The House model matches snapshot when object is created with IRawHouseData info 1`] = `
Object {
  "address": "Paseo de la Castellana",
  "city": "madrid",
  "id": "79906",
  "image": "https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/74ef00db848c1ce44ea03677836b5bb61243e3f9d6a9d5a656d3c3cb.jpg",
  "link": "https://www.spotahome.com/madrid/for-rent:apartments/79906?utm_source=housinganywhere&utm_medium=cffeeds&utm_campaign=normalads",
  "title": "madrid - 280.000,00 €",
}
`;
